#!/usr/bin/env python3
# coding=utf-8

import sys
from PyQt5 import QtCore, QtWidgets, uic, QtGui
from PyQt5.QtGui import QPixmap
from PyQt5.QtWidgets import QTableWidgetItem, QAbstractItemView

answers = ['', '', '']  # 1 - form2, 2 - form3, 3 - form5


class Form1(QtWidgets.QMainWindow):
    # аргумент str говорит о том, что сигнал должен быть сторокового типа
    switch_window = QtCore.pyqtSignal(str)

    def __init__(self):
        super(Form1, self).__init__()
        uic.loadUi('uis/form1.ui', self)

        self.setWindowIcon(QtGui.QIcon('_MY_PICTURES/logo.png'))

        self.qPushButtonNext1.clicked.connect(self.qPushButtonNext1_onClick)

    def qPushButtonNext1_onClick(self):
        self.switch_window.emit('1>2')


class Form2(QtWidgets.QMainWindow):
    switch_window = QtCore.pyqtSignal(str)

    def __init__(self):
        super(Form2, self).__init__()
        uic.loadUi('uis/form2.ui', self)

        self.setWindowIcon(QtGui.QIcon('_MY_PICTURES/logo.png'))

        self.qPushButtonBack2.clicked.connect(self.qPushButtonBack2_onClick)
        self.qPushButtonNext2.clicked.connect(self.qPushButtonNext2_onClick)


    def qPushButtonBack2_onClick(self):
        self.switch_window.emit('1<2')

    def qPushButtonNext2_onClick(self):
        self.switch_window.emit('2>3')


class Form3(QtWidgets.QMainWindow):
    switch_window = QtCore.pyqtSignal(str)

    def __init__(self):
        super(Form3, self).__init__()
        uic.loadUi('uis/form3.ui', self)

        self.setWindowIcon(QtGui.QIcon('_MY_PICTURES/logo.png'))

        self.qPushButtonBack3.clicked.connect(self.qPushButtonBack3_onClick)
        self.qPushButtonNext3.clicked.connect(self.qPushButtonNext3_onClick)


    def qPushButtonBack3_onClick(self):
        self.switch_window.emit('2<3')

    def qPushButtonNext3_onClick(self):
        self.switch_window.emit('3>4')


class Form4(QtWidgets.QMainWindow):
    switch_window = QtCore.pyqtSignal(str)

    def __init__(self):
        super(Form4, self).__init__()
        uic.loadUi('uis/form4.ui', self)

        self.setWindowIcon(QtGui.QIcon('_MY_PICTURES/logo.png'))

        # self.tableWidget.setItem(0, 1, QTableWidgetItem(answers[0]))

        self.qPushButtonBack4.clicked.connect(self.qPushButtonBack4_onClick)
        self.qPushButtonNext4.clicked.connect(self.qPushButtonNext4_onClick)

    def qPushButtonBack4_onClick(self):
        self.switch_window.emit('3<4')

    def qPushButtonNext4_onClick(self):
        self.switch_window.emit('4>5')


class Form5(QtWidgets.QMainWindow):
    switch_window = QtCore.pyqtSignal(str)

    def __init__(self):
        super(Form5, self).__init__()
        uic.loadUi('uis/form5.ui', self)

        self.setWindowIcon(QtGui.QIcon('_MY_PICTURES/logo.png'))

        self.qPushButtonBack5.clicked.connect(self.qPushButtonBack5_onClick)
        self.qPushButtonExit5.clicked.connect(self.close)

    def qPushButtonBack5_onClick(self):
        self.switch_window.emit("4<5")


'''
Класс управления переключения окон
'''


class Controller:
    def __init__(self):
        pass

    def select_forms(self, text):
        if text == '1':
            self.form1 = Form1()
            self.form1.switch_window.connect(self.select_forms)
            self.form1.show()

        if text == '1>2':
            self.form2 = Form2()
            self.form2.switch_window.connect(self.select_forms)
            self.form2.show()
            self.form1.close()

        if text == '2>3':
            self.form3 = Form3()
            self.form3.switch_window.connect(self.select_forms)
            self.form3.show()
            self.form2.close()

        if text == '3>4':
            self.form4 = Form4()
            self.form4.switch_window.connect(self.select_forms)
            self.form4.show()
            self.form3.close()

        if text == '4>5':
            self.form5 = Form5()
            self.form5.switch_window.connect(self.select_forms)
            self.form5.show()
            self.form4.close()

        if text == '4<5':
            self.form4 = Form4()
            self.form4.switch_window.connect(self.select_forms)
            self.form4.show()
            self.form5.close()

        if text == '3<4':
            self.form3 = Form3()
            self.form3.switch_window.connect(self.select_forms)
            self.form3.show()
            self.form4.close()

        if text == '2<3':
            self.form2 = Form2()
            self.form2.switch_window.connect(self.select_forms)
            self.form2.show()
            self.form3.close()

        if text == '1<2':
            self.form1 = Form1()
            self.form1.switch_window.connect(self.select_forms)
            self.form1.show()
            self.form2.close()


def main():
    app = QtWidgets.QApplication(sys.argv)
    controller = Controller()
    controller.select_forms("1")
    sys.exit(app.exec_())


if __name__ == '__main__':
    main()
